### PR TITLE
Fix undefined order of waypoint listener config

### PR DESCRIPTION
**Pull request by hanxiaop**
**Please provide a description of this PR:**
Currently, the order of waypoint listeners' config is changing instead of a consistent list.

### CODE REVIEW

File: pilot/pkg/networking/core/v1alpha3/waypoint.go [hunk 1]
```diff
@@ -38,27 +40,40 @@ const (
...
+				if ws.services == nil {
+					ws.services = map[host.Name]*model.Service{}
+				}
+				ws.services[svc.Hostname] = svc
 			}
 		}
 	}
-	return svcs
+	keys := maps.Keys(ws.services)
+	slices.Sort(keys)
```
Comment (hzxuzhonghu) [inline pilot/pkg/networking/core/v1alpha3/waypoint.go:72]: nit: sort by name doesn't seem right, keep consistent with other places by creation timestamp

Comment (hanxiaop) [reply pilot/pkg/networking/core/v1alpha3/waypoint.go]: Thanks, updated. Just realized that the logic for sorting the workload was missing as well, which resulted in an undefined order of the custom IP match list. I've added a similar sorting logic to the workload.

File: pilot/pkg/serviceregistry/kube/controller/ambientindex_external.go [hunk 2]
```diff
@@ -119,6 +120,7 @@ func (a *AmbientIndexImpl) handleServiceEntry(svcEntry *apiv1alpha3.ServiceEntry
 		var wl *model.WorkloadInfo
 		if event != model.EventDelete {
 			wl = a.extractWorkloadEntrySpec(we, svcEntry.GetNamespace(), svcEntry.GetName(), svcEntry, c)
+			handleWorkloadCreationTime(oldWl, wl)
```
Comment (hzxuzhonghu) [inline pilot/pkg/serviceregistry/kube/controller/ambientindex_external.go:123]: why don't directly set it to SE.CreationTimestamp

Comment (hanxiaop) [reply pilot/pkg/serviceregistry/kube/controller/ambientindex_external.go]: You right, this can just be set to se's creation time.

**Review by hzxuzhonghu — APPROVED** [review]
LGTM. Though ambient index may need some enhancement to speed up searching

**Pull request merged**